src/node/internal/internal_fs_callback.ts

```diff
@@ -534,16 +535,34 @@ export function mkdir(
 
 export function mkdtemp(
   prefix: FilePath,
-  optionsOrCallback: SingleArgCallback<string> | MkdirTempSyncOptions,
+  optionsOrCallback:
+    | SingleArgCallback<string>
+    | MkdirTempSyncOptions
+    | BufferEncoding
+    | 'buffer'
+    | null,
   callback?: SingleArgCallback<string>
 ): void {
-  let options: MkdirTempSyncOptions;
+  let options: MkdirTempSyncOptions | ValidEncoding;
   if (typeof optionsOrCallback === 'function') {
     callback = optionsOrCallback;
     options = {};
   } else {
     options = optionsOrCallback;
   }
+  if (typeof options === 'string' || options == null) {
+    options = { encoding: options };
+  }
+  validateObject(options, 'options');
+  const { encoding = null } = options;
+  if (
+    encoding !== null &&
```

Comment:
There is a validateEncoding method if I remember correctly.

Reply:
Which doesn't handle this. 

```
export function validateEncoding(
  data: unknown,
  encoding: string
): asserts data is string {
  const normalizedEncoding = normalizeEncoding(encoding);
  const length = (data as any).length; // eslint-disable-line

  if (normalizedEncoding === bufferUtil.HEX && length % 2 !== 0) {
    throw new ERR_INVALID_ARG_VALUE(
      'encoding',
      encoding,
      `is invalid for data of length ${length}`
    );
  }
}
```